core/shared/src/main/scala/zio/prelude/Inverse.scala

```diff
@@ -22,6 +24,16 @@ import zio.test.laws.{ Lawful, Laws }
  */
 trait Inverse[A] extends Identity[A] {
   def inverse(l: => A, r: => A): A
+  def multiply(n: Int)(a: A): A =
+    (multiplyOption(n)(a)).getOrElse(identity)
+  override def multiplyOption(n: Int)(a: A): Option[A] = {
+    @tailrec
+    def multiplyHelper(res: A, n: Int): Option[A] =
+      if (n <= 0) None
+      else if (n == 1) Some(res)
+      else multiplyHelper(inverse(a, res), n - 1)
```

Comment:
how about
```scala
      if (n = 0) identity
      else if (n < 0) multiplyHelper(inverse(res, a), n + 1)
      else multiplyHelper(combine(res, a), n + 1)
```
`multiplyHelper` could return plain `A`s and `multiplyOption` could return `Some` instead of `Option`